app/code/Magento/Cms/view/frontend/layout/cms_page_view.xml

```diff
@@ -9,6 +9,7 @@
     <body>
         <referenceContainer name="content">
             <block class="Magento\Cms\Block\Page" name="cms_page"/>
+            <block class="Magento\Catalog\Block\Product\AbstractProduct" name="widget.add.product.to.cart" template="Magento_Cms::widget/product/addToCart.phtml"/>
```

Comment:
Hi @Usik2203,
You add this block to **all** CMS pages, regardless of the Product List Widget usage. This should be done only when the specified widget is used on the page, IMO.
Maybe a better place for such changes is `app/code/Magento/Catalog/view/frontend/templates/product/widget/new/content/new_list.phtml` template, for example.

Reply:
Also, the class `Magento\Catalog\Block\Product\AbstractProduct` seems to be heavy, having a look at its constructor class.

Reply:
Hi @vasilii-b 
I have applied new solution.
Thank You a lot for Your advice.